core/src/main/java/org/jdbi/v3/core/qualifier/Qualifiers.java

```diff
@@ -36,40 +38,87 @@
 @Beta
 public class Qualifiers implements JdbiConfig<Qualifiers> {
     private static final JdbiCache<AnnotatedElement[], Set<Annotation>> QUALIFIER_CACHE = JdbiCaches.declare(
-            elements -> elements.length == 1 ? elements[0] : new HashSet<>(Arrays.asList(elements)),
-            (Function<AnnotatedElement[], Set<Annotation>>) Qualifiers::getQualifiers);
+        elements -> elements.length == 1 ? elements[0] : new HashSet<>(Arrays.asList(elements)),
+        (registry, elements) -> registry.get(Qualifiers.class).resolveQualifiers(elements));
+    private final Set<Function<AnnotatedElement, Set<Annotation>>> resolvers = new CopyOnWriteArraySet<>();
     private ConfigRegistry registry;
 
-    public Qualifiers() {}
+    public Qualifiers() {
+        resolvers.add(Qualifiers::getQualifierAnnotations);
+        resolvers.add(Qualifiers::inspectQualifiedCarrier);
+    }
+
+    private Qualifiers(Qualifiers other) {
+        this.resolvers.addAll(other.resolvers);
+        this.registry = null;
+    }
 
     @Override
     public void setRegistry(ConfigRegistry registry) {
         this.registry = registry;
     }
 
+    /**
+     * Adds a qualifier resolver, which inspects annotated elements (e.g. methods or parameters)
+     * and returns a set of qualifiers that are found to belong on it. An example resolver is {@link #getQualifierAnnotations},
+     * which searches for annotations themselves annotated with {@link Qualifier}.
+     *
+     * Resolvers should return static results that don't vary over time; therefore their results are cached.
+     * Adding a resolver will clear the local qualifier resolution cache, but is currently not guaranteed to have any effect on {@code SqlObjects}.
```

Comment:
I think this is a little misleading: as long as you add the resolvers before you initialize your SqlObjects, it should work just fine with SqlObject.

Reply:
Guess I left out the word "_this_ is not guaranteed to _retroactively_ affect sqlobjects"